.github/workflows/release.yml

```diff
@@ -0,0 +1,51 @@
+# This GitHub action can publish assets for release when a tag is created.
+# Currently its setup to run on any tag that matches the pattern "v*" (ie. v0.1.0).
+#
+# This uses an action (crazy-max/ghaction-import-gpg) that assumes you set your
+# private key in the `GPG_PRIVATE_KEY` secret and passphrase in the `PASSPHRASE`
+# secret. If you would rather own your own GPG handling, please fork this action
+# or use an alternative one for key handling.
+#
+# You will need to pass the `--batch` flag to `gpg` in your signing step 
+# in `goreleaser` to indicate this is being used in a non-interactive mode.
+#
+name: release
+on:
+  push:
+    tags:
+      - 'v*'
+permissions:
+  contents: write
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      -
+        name: Unshallow
+        run: git fetch --prune --unshallow
+      -
+        name: Set up Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+      -
+        name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@111c56156bcc6918c056dbef52164cfa583dc549 # v5.2.0
+        id: import_gpg
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.PASSPHRASE }}
```

Comment:
@RutvikS-crest I've added the GPG key and passphrase as GitHub secrets, matching the above values. Can you confirm whether this workflow requires any further modification or testing, now that the keys are in place?

Reply:
That's it from repo side, I am assuming the key is also registered with Cisco/DevNet space in Hashicorp registry.

Reply:
@lelia , can you confirm Rutvik's assumption is correct?

Reply:
@rquedas @RutvikS-crest it should now be registered both with Hashicorp and GitHub, yes. There is a `cisco-service` account (shared between CiscoDevNet and a few other Cisco open source orgs) which we have used to upload the new keys to.

Reply:
@RutvikS-crest FYI that I'm going to remove the repository-level placeholders for `GPG_PRIVATE_KEY` and `PASSPHRASE`, as they are currently superseding the organization-level secrets that I've added.